aspects/aspects.go

```diff
@@ -650,30 +707,83 @@ func NewJSONDataBag() JSONDataBag {
 // Get takes a path and a pointer to a variable into which the value referenced
 // by the path is written. The path can be dotted. For each dot a JSON object
 // is expected to exist (e.g., "a.b" is mapped to {"a": {"b": <value>}}).
-func (s JSONDataBag) Get(path string, value interface{}) error {
+func (s JSONDataBag) Get(path string, value *interface{}) error {
 	subKeys := strings.Split(path, ".")
 	return get(subKeys, 0, s, value)
 }
 
-func get(subKeys []string, index int, node map[string]json.RawMessage, result interface{}) error {
+func get(subKeys []string, index int, node map[string]json.RawMessage, result *interface{}) error {
 	key := subKeys[index]
+	matchAll := key[0] == '{' && key[len(key)-1] == '}'
+
 	rawLevel, ok := node[key]
-	if !ok {
+	if !matchAll && !ok {
 		pathPrefix := strings.Join(subKeys[:index+1], ".")
-		return PathNotFoundError(fmt.Sprintf("no value was found under path %q", pathPrefix))
+		return pathErrorf("no value was found under path %q", pathPrefix)
 	}
 
 	// read the final value
 	if index == len(subKeys)-1 {
+		if matchAll {
+			// request ends in placeholder so return map to all values (but unmarshal first)
+			level := make(map[string]interface{}, len(node))
+			for k, v := range node {
+				var deser interface{}
+				if err := json.Unmarshal(v, &deser); err != nil {
+					return err
+				}
+				level[k] = deser
+			}
+
+			*result = level
+			return nil
+		}
+
 		err := json.Unmarshal(rawLevel, result)
 		if uErr, ok := err.(*json.UnmarshalTypeError); ok {
 			path := strings.Join(subKeys, ".")
-			return fmt.Errorf("cannot read value of %q into %T: maps to %s", path, result, uErr.Value)
+			return pathErrorf("cannot read value of %q into %T: maps to %s", path, result, uErr.Value)
```

Comment:
I notice that tests don't reach this error case

Reply:
This is no longer possible since we unmarshal into a local interface{} variable instead of something coming from out of the package. I've simplified it. Thanks